### PR TITLE
Get rid of allowedsecretslist from app KV

### PR DESCRIPTION
**Pull request by WaelAbuSeada**
#### Summary 
Get rid of allowedsecretslist from app KV

Fixes #
[AB#544481](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/544481)

### CODE REVIEW

File: src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al [hunk 1]
```diff
@@ -23,44 +23,24 @@ codeunit 2202 "Azure Key Vault Impl."
...
     begin
         // Gets the secret as a Text from the key vault, given a SecretName.
-
-        if not InitializeAllowedSecretNames() then
-            Error(InitializeAllowedSecretNamesErr);
-
-        if not IsSecretNameAllowed(SecretName) then
-            Error(SecretNotFoundErr, SecretName);
-
         Secret := GetSecretFromClient(SecretName);
```
Comment (Groenbech96) [inline src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al:38]: Clean up this function, as the only thing it does is call another function.

File: src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al [hunk 2]
```diff
@@ -23,44 +23,24 @@ codeunit 2202 "Azure Key Vault Impl."
...
     [NonDebuggable]
     procedure GetAzureKeyVaultSecret(SecretName: Text; var Secret: SecretText)
     begin
-        if not InitializeAllowedSecretNames() then
-            Error(InitializeAllowedSecretNamesErr);
-
-        if not IsSecretNameAllowed(SecretName) then
-            Error(SecretNotFoundErr, SecretName);
-
         Secret := GetSecretFromClient(SecretName);
```
Comment (Groenbech96) [inline src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al:44]: Same as above.

**Review by Groenbech96 — CHANGES_REQUESTED** [review]
Some comments, but otherwise ok

File: src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al (hunk 1 above)
Comment (WaelAbuSeada) [reply src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al:38]: Yea, but this is called twice from a method that returns text and secret text

**Review by Aleyenda — DISMISSED** [review]

**Review by Groenbech96 — DISMISSED** [review]

**Review by darjoo — APPROVED** [review]

**Review by nikolakukrika — APPROVED** [review]

**Pull request merged**